test/dtest/conftest.py

```diff
@@ -0,0 +1,204 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import argparse
+import logging
+import re
+import warnings
+from collections import defaultdict
+
+import pytest
+from packaging.version import Version
+
+from test.dtest.dtest_config import DTestConfig
+from test.dtest.dtest_setup import DTestSetup
+from test.dtest.dtest_setup_overrides import DTestSetupOverrides
+from test.dtest.tools.marks import IssueClosed, MarkedLocals, RequirePredicate, enable_with_features, get_version
+from test.dtest.tools.misc import colored_text
+
+
+pytest_plugins = [
+    "test.topology.conftest",
+]
+
+logger = logging.getLogger(__name__)
+
+
+def pytest_addoption(parser):
+    parser.addoption("--use-vnodes", action="store_true", default=True, help="Determines wither or not to setup clusters using vnodes for tests")
+    parser.addoption("--num-tokens", action="store", default=256, help="Number of tokens to set num_tokens yaml setting to when creating instances " "with vnodes enabled")
+    parser.addoption("--experimental-features", type=lambda s: s.split(","), action="store", help="Pass experimental features <feature>,<feature> to enable")
+    parser.addoption("--tablets", action=argparse.BooleanOptionalAction, default=False, help="Whether to enable tablets support (default: %(default)s)")
+    parser.addoption("--force-gossip-topology-changes", action="store_true", default=False, help="force gossip topology changes in a fresh cluster")
+
+
+@pytest.fixture(scope="function", autouse=True)
+def fixture_dtest_setup_overrides(dtest_config):
+    """
+    no-op default implementation of fixture_dtest_setup_overrides.
+    we run this when a test class hasn't implemented their own
+    fixture_dtest_setup_overrides
+    """
+    return DTestSetupOverrides()
+
+
+@pytest.fixture(scope="function", autouse=False)
+def fixture_dtest_setup(request, dtest_config, fixture_dtest_setup_overrides, manager):
```

Comment:
We want to port **tests** to the test.py framework

this looks like you're porting the **dtest framework** instead?